Class Presentations/reveal-presentations/ObjectsAndDom.html

```diff
@@ -0,0 +1,604 @@
+				<h2>Objects</h2>
+				<p>"A collection of named values"</p>
+				<p>"Variables containing variables"</p>
+				<p>"The building blocks of JavaScript"</p>
+				<p>(source: W3Schools)</p>
+			</section>
+
+			<section>
+				<p>We use objects to store and access information.
+					They are a way to represent data.
+				We can even use them to represent real life objects.</p>
+			</section>
+
+			<section>
+				<p>We're going to create a representative of an object the same way
+					 it's done in JavaScript. To help us, we have a special guest...</p>
+				<img class="fragment" src="../presentation-images/objectsanddom/Florida_Box_Turtle_Digon3a.jpg"
+				width="400" height="300">
+				<p class="fragment">Kelly, the common box turtle!</p>
+			</section>
+
+			<section>
+				<p>Remember when we talked about properties and values? JavaScript objects
+					have those too, and so does Kelly.</p>
+					<ol>
+						<li>weight: 2.4</li>
+						<li>gender: female</li>
+						<li>...</li>
+					</ol>
+			</section>
+
+			<section>
+				<img class="fragment" src="../presentation-images/objectsanddom/Florida_Box_Turtle_Digon3a.jpg"
+				width="400" height="300">
+				<p class="fragment"> Take 60 seconds to come up with as many other
+						properties as you can.</p>
+				<p class="fragment"> Ready? GO!</p>
+			</section>
+
+			<section>
+				<p>Now, how do we create Kelly with these properties?</p>
+			</section>
+
+			<section>
+				<h2>Goals</h2>
+				<ul class="checklist">
+					<li>
+						<input type="checkbox" id="goal1" checked />
+						<label for="goal1">What are objects?</label>
+					</li>
+					<li>
+						<input type="checkbox" id="goal2"/>
+						<label for="goal2">Creating Objects</label>
+					</li>
+					<li>
+						<input type="checkbox" id="goal3"/>
+						<label for="goal3">Methods</label>
+					</li>
+					<li>
+						<input type="checkbox" id="goal4"/>
+						<label for="goal4">Browser Object Model</label>
+					</li>
+					<li>
+						<input type="checkbox" id="goal5"/>
+						<label for="goal5">The Document Object Model</label>
+					</li>
+				</ul>
+			</section>
+
+			<section>
+				<h2>Creating Objects</h2>
+				<h3>feat. Kelly the Box Turtle</h3>
+			</section>
+
+			<section data-markdown>
+				```javascript
+				var kelly = {
+					species: "common box turtle",
+					weight: 2.4,
+					gender: female,
+					quote: "I'll get there when I get there"
+				}
+				```
+				(You can see how this is a collection of variables.)
+			</section>
+
+			<section>
+				<section>
+					<h2>There are two ways to access information about Kelly:</h2>
+					<h3>(scroll down)</h3>
+				</section>
+				<section>
+					<p>1. Using periods (dot notation).</p>
+				</section>
+				<section data-markdown>
+					```javascript
+					var text = kelly.species;
+					```
+				</section>
+				<section>
+					<p>2. Using brackets (bracket notation).</p>
+				</section>
+				<section data-markdown>
+					```javascript
+					var text = kelly[species];
```

Comment:
So, one thing about bracket notation is that the brackets need to contain a string that is the same as the key of the `key: value` pair that you're trying to access.

For example:
```
var text = kelly["species"];     // "common box turtle"

var lookup = "species";
var text = kelly[lookup];        // "common box turtle"

// So, you can see that the above are two different ways of doing the same thing...

var text = kelly[species]   // Probably  a reference error, since species === undefined

```